ivadomed/main.py

```diff
@@ -362,185 +414,381 @@ def run_command(context, n_gif=0, thr_increment=None, resume_training=False, no_
+    #  Loader Version Determination:
+    #################################
+    # If the new keywords are present, we use the new loader pathway. Ignoring all "contrast_params" keys etc fro now.
+    if context.get(DataloaderKW.DATASET_GROUPS) and context.get(DataloaderKW.EXPECTED_GT) and context.get(DataloaderKW.EXPECTED_INPUT):
+        loader_version: str = LoaderParamsKW.MULTI_PATH_LOADER
+    else:
+        loader_version: str = LoaderParamsKW.TRADITIONAL_BIDS_LOADER
+        logger.info(f"Using {loader_version} loader version because new V2 loader keywords are not present in the context.")
+
+    #################################
+
+    # If the older loader parameters key, we default to use the old loader pathway relying on bids.
     path_output = set_output_path(context)
-    path_log = Path(context.get('path_output'), context.get('log_file'))
+    path_log: Path = Path(
+        context.get('path_output'),
+        context.get('log_file')
+    )
     logger.remove()
     logger.add(str(path_log))
     logger.add(sys.stdout)
 
     # Create a log with the version of the Ivadomed software and the version of the Annexed dataset (if present)
-    create_dataset_and_ivadomed_version_log(context)
+    # create_dataset_and_ivadomed_version_log(context)
 
     cuda_available, device = imed_utils.define_device(context[ConfigKW.GPU_IDS][0])
 
     # BACKWARDS COMPATIBILITY: If bids_path is string, assign to list - Do this here so it propagates to all functions
-    context[ConfigKW.LOADER_PARAMETERS][LoaderParamsKW.PATH_DATA] =\
-        imed_utils.format_path_data(context[ConfigKW.LOADER_PARAMETERS][LoaderParamsKW.PATH_DATA])
+    context[ConfigKW.LOADER_PARAMETERS][LoaderParamsKW.PATH_DATA] = imed_utils.format_path_data(
+            context[ConfigKW.LOADER_PARAMETERS][LoaderParamsKW.PATH_DATA]
+        )
 
-    # Loader params
-    loader_params = set_loader_params(context, command == "train")
+    # Set Loader params leveraging information from Loader_Parameters for the BIDS Loader version.
+    # Not needed in the alternative loader version.
+    loader_params: dict = update_loader_params(context, command == "train", loader_version)
 
-    # Get transforms for each subdataset
-    transform_train_params, transform_valid_params, transform_test_params = \
-        imed_transforms.get_subdatasets_transforms(context[ConfigKW.TRANSFORMATION])
+    # Get transforms dictionaries for each subdataset
+    transform_train_params, \
+    transform_valid_params, \
+    transform_test_params = imed_transforms.get_subdatasets_transforms(context[ConfigKW.TRANSFORMATION])
 
-    # MODEL PARAMETERS
+    # Update and populate model parameters AND loader parameters
     model_params, loader_params = set_model_params(context, loader_params)
 
-    if command == 'segment':
+    if command == CommandKW.SEGMENT and loader_version == LoaderParamsKW.TRADITIONAL_BIDS_LOADER:
         run_segment_command(context, model_params, no_patch, overlap_2d)
```

Comment:
Just a heads-up, the `--segment` command also relies heavily on BIDS (but is less critical than the `--test` command because we can more easily use the `segment_volume `API instead).

As a reference, we were already thinking about revamping this here: https://github.com/ivadomed/ivadomed/issues/1002

Reply:
Same as above. This is another significant part of the FileDataset changes I think we should undertake before BIDSDataframe changes.